crates/bcr-ebill-core/src/protocol/mod.rs

```diff
@@ -115,17 +148,18 @@ fn get_version(_event_type: &EventType) -> String {
 pub struct EventEnvelope {
     pub event_type: EventType,
     pub version: String,
-    pub data: Value,
+    pub data: String,
```

Comment:
Why `String` and not `Vec<u8>` here?

I'm just asking because in `BillBlock` I'll change the `String` to `Vec<u8>` and only encode it for persistence, which should save around ~30% of data size for the (potentially big) block data.

Is there a reason that would prevent us from doing that here as well? Since when being sent, the whole `Event` is serialized, encoded and encrypted anyway, right?

Reply:
Hmm :thinking: 

Do you mean the `nostr::Event`? I thought there it's in the `content` which is encrypted+encoded (for public chain events) and plaintext I guess (for direct msgs, because encryption is on Nostr side) and from there, based on the type, we unwrap to the corresponding event type.

But since this `content` is just a string, couldn't that just be a base58 encoded borsh-serialized, or encrypted Vec<u8>? Or is there something on Nostr-side limiting us here?

They way I thought about it is, that whatever is the "content" of the messages from nostr doesn't contain any JSON on our side anymore. Am I missing something? :thinking: